### PR TITLE
Add/new email comparison card workspace wire

### PR DESCRIPTION
**Pull request by AllTerrainDeveloper**
#### Changes proposed in this Pull Request

The changes in this pull request are allowing the Google Workspace Card to add the product to the shopping cart.

#### Testing instructions

1. Go to a site with domains without email subscription
2. Click on 'Add email' for the desired site
3. Once you are in the comparison page, add at the end the following flag: `?flags=emails/new-email-comparison`
4. Try to fill form data and check that the item is added into the cart and redirected to the checkout.

image

Related to {1200182182542585-as-1201457616446698}

### CODE REVIEW

File: client/my-sites/email/email-providers-stacked-comparison/provider-cards/professional-email-card.tsx [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+		translate( '24/7 support via email' ),
+	];
+};
+
+const professionalEmailFormattedPrice = (
+	formattedPriceClassName: string,
+	cost: number,
+	currencyCode: string
+) => {
+	return translate( '{{price/}} /mailbox', {
```
Comment (a8ci18n) [inline client/my-sites/email/email-providers-stacked-comparison/provider-cards/professional-email-card.tsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **18** times:
`translate( '{{price/}} /mailbox /month' )` *ES Score: 8*

File: client/my-sites/email/email-providers-stacked-comparison/provider-cards/professional-email-card.tsx [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+
+import './professional-email-card.scss';
+
+// eslint-disable-next-line @typescript-eslint/no-empty-function
+const noop = () => {};
+const logo = <Gridicon className="professional-email-card__logo" icon="my-sites" />;
+const badge = <img src={ poweredByTitanLogo } alt={ translate( 'Powered by Titan' ) } />;
+const getTitanFeatures = () => {
+	return [
+		translate( 'Inbox, calendars, and contacts' ),
```
Comment (a8ci18n) [inline client/my-sites/email/email-providers-stacked-comparison/provider-cards/professional-email-card.tsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **20** times:
`translate( 'Email, calendars, and contacts' )` *ES Score: 6*

Comment (stephanethomas) [reply client/my-sites/email/email-providers-stacked-comparison/provider-cards/professional-email-card.tsx]: It's probably better to keep the string that is already translated.

File: client/my-sites/email/email-providers-stacked-comparison/provider-cards/google-workspace-card.tsx [hunk 3]
```diff
@@ -158,8 +166,60 @@ const GoogleWorkspaceCard: FunctionComponent< EmailProvidersStackedCardProps > =
 		</div>
 	);
 
-	const onGoogleConfirmNewMailboxes = noop;
-	const onGoogleUsersChange = noop;
+	const [ googleUsers, setGoolgeUsers ] = useState( newUsers( selectedDomainName ) );
```
Comment (stephanethomas) [inline client/my-sites/email/email-providers-stacked-comparison/provider-cards/google-workspace-card.tsx:169]: Typo 😉:
```suggestion
	const [ googleUsers, setGoogleUsers ] = useState( newUsers( selectedDomainName ) );
```

**Review by stephanethomas — APPROVED** [review]
This looks good and works fine. This is unrelated to your change but I've noticed some weird display issues with the check icons from the list of features (they don't have all the same appearance):

image

**Comment by stephanethomas** [comment]
Hey @saygunnyc 👋. Just a thought: the way we use that select box seems a bit weird to me, what do you think of replacing the `Learn more` select box with a link? This would also save some space on mobile devices:

Before | After
------ | -----
image | image

**Comment by AllTerrainDeveloper** [comment]
I've changed it also in the next PR. It's smaller without button.

File: client/my-sites/email/email-providers-stacked-comparison/provider-cards/google-workspace-card.tsx (hunk 3 above)
Comment (AllTerrainDeveloper) [reply client/my-sites/email/email-providers-stacked-comparison/provider-cards/google-workspace-card.tsx:169]: 😄 Next pull request addressed it [here](https://github.com/Automattic/wp-calypso/pull/58761/files#diff-9d9202fc059e9bb4ee4a671444405c105288c06253b359ca52a4364647f60d93R145)

**Comment by AllTerrainDeveloper** [comment]
@stephanethomas @saygunnyc I am missing the Titan Badge (and some margins), I'll add it tomorrow morning.
mobile support expanded

**Comment by saygunnyc** [comment]
> Hey @saygunnyc 👋. Just a thought: the way we use that select box seems a bit weird to me, what do you think of replacing the `Learn more` select box with a link? This would also save some space on mobile devices:
> 
> Before	After
> image	image

Hey @stephanethomas , I just wanted to point you in the direction of this Figma file. – https://www.figma.com/file/IKskpfdoC4OplHAvLihFCZ/1.Onboarding---Email-Onboarding-i2?node-id=923%3A25362

Below I'm sharing mobile designs for visibility.

**Collapsed State** (**Show all features**)


**Expanded State**

**Comment by saygunnyc** [comment]
> @stephanethomas @saygunnyc I am missing the Titan Badge (and some margins), I'll add it tomorrow morning. mobile support expanded

Hey @AllTerrainDeveloper !
Great to see progress on this! As mentioned over here – pcYYuD-62-p2#comment-143 these are currently quite far off from the intended designs. _– Column sizes, icon sizes, spacings, input styles, etc_  I believe most of these correct styles should already be available over the new onboarding designs that are in production. Perhaps that could be helpful.

Other than that is there a way for me to help?

**Pull request merged**